open_fdd/desktop/services/ingest_service.py

```
@@ -0,0 +1,98 @@
+            "dropped_rows": dropped_rows,
+        }
+
+    def ingest_weather(self, *, site_id: str, days_back: int = 1) -> dict[str, Any]:
+        result = run_weather_fetch(store=self.feather_store, site_id=site_id, days_back=days_back)
+        return {"rows": result.rows, "source": result.source}
+
+    def ingest_onboard(self, *, site_id: str) -> dict[str, Any]:
+        result = run_onboard_scrape(store=self.feather_store, site_id=site_id)
+        return {"rows": result.rows, "source": "onboard"}
+
+    def load_source_frame(self, *, source: str, site_id: str) -> pd.DataFrame:
```

Ruff (0.15.12)

[error] Undefined name `pd`

(F821)